pkg/cosign/cosign.go

```diff
@@ -145,23 +147,32 @@ func getFulcioRoots(roots []byte) (*x509.CertPool, error) {
 
 // FetchAttestations retrieves signed attestations and decodes them into in-toto statements
 // https://github.com/in-toto/attestation/blob/main/spec/README.md#statement
-func FetchAttestations(imageRef string, key string, repository string, log logr.Logger) ([]map[string]interface{}, error) {
+func FetchAttestations(imageRef string, imageVerify *v1.ImageVerification, log logr.Logger) ([]map[string]interface{}, error) {
 	ctx := context.Background()
-	var pubKey signature.Verifier
 	var err error
 
-	if strings.HasPrefix(key, "-----BEGIN PUBLIC KEY-----") {
-		pubKey, err = decodePEM([]byte(key))
-		if err != nil {
-			return nil, errors.Wrap(err, "decode pem")
+	cosignOpts := &cosign.CheckOpts{
+		ClaimVerifier: cosign.IntotoSubjectClaimVerifier,
+	}
+
+	if imageVerify.Key != "" {
+		if strings.HasPrefix(imageVerify.Key, "-----BEGIN PUBLIC KEY-----") {
+			cosignOpts.SigVerifier, err = decodePEM([]byte(imageVerify.Key))
+		} else {
+			cosignOpts.SigVerifier, err = sigs.PublicKeyFromKeyRef(ctx, imageVerify.Key)
 		}
 	} else {
-		pubKey, err = sigs.PublicKeyFromKeyRef(ctx, key)
-		if err != nil {
-			return nil, errors.Wrap(err, "loading public key")
+		cosignOpts.CertEmail = ""
```

Comment:
```suggestion
```

I believe we don't need to explicitly set the CertEmail to an empty string. It defaults to an empty string. https://github.com/sigstore/cosign/blob/f436d7637caaa9073522ae65a8416e38cd69c4f2/pkg/cosign/verify.go#L78

Reply:
Yes, its not really necessary. Should add some comments that we explicitly do not set the email here, as we want to do a wildcard match later. Will update separately.